tests/pytests/functional/states/pkgrepo/test_debian.py

```diff
@@ -465,3 +465,30 @@ def _run(arch=None, test=False):
         assert not ret.changes
         assert "already" in ret.comment
         assert ret.result is True
+
+
+@pytest.mark.requires_salt_states("pkgrepo.managed", "pkgrepo.absent")
+def test_repo_present_absent_trailing_slash_uri(pkgrepo):
+    """
+    test adding a repo with a trailing slash in the uri
+    """
+    repo_file = "/etc/apt/sources.list.d/trailing-slash.list"
+    repo_content = "deb http://www.deb-multimedia.org/ stable main"
+    # initial creation
+    ret = pkgrepo.managed(
+        name=repo_content, file=repo_file, refresh=False, clean_file=True
+    )
+    with salt.utils.files.fopen(repo_file, "r") as fp:
+        file_content = fp.read()
+    assert file_content.strip() == "deb http://www.deb-multimedia.org stable main"
+    assert ret.changes
+    # no changes
+    ret = pkgrepo.managed(name=repo_content, file=repo_file, refresh=False)
+    assert not ret.changes
+    # absent
+    ret = pkgrepo.absent(name=repo_content)
+    assert ret.result
+    try:
```

Comment:
The repo_file will not get cleaned up if the test fails before this try except. Can you either put the entire test in a try finally, or my preferred method, use a pytest fixture to ensure it gets cleaned up after the test.